### PR TITLE
feat(wmg): Improve delete gene discoverability in Gene Expression

### PR DESCRIPTION
**Pull request by ashin-czi**
## Reason for Change

- #3304 

## Changes

- add
    - New button icon above gene name to delete individual genes. Delete icon is visible on hover for each gene.
- remove
- modify

## Testing steps

rdev: https://delete-gene-frontend.rdev.single-cell.czi.technology/gene-expression

### CODE REVIEW

**Comment by ashin-czi** [comment]
converting back to draft to add analytics:

https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/gh/chanzuckerberg/single-cell-data-portal/4230

**Review by seve — COMMENTED** [review]
I think there's a simpler way to handle the hiding/un-hiding without triggering re-renders. You can add a hover style capture in the parent component that modifies the child.
I imagine it would look something like this:
```css
.child-delete-button-class {
    visibility: hidden;
}
.parent-gene-button-class:hover .child-delete-button-class {
    visibility: visible;
 }

```

File: frontend/src/views/WheresMyGene/common/store/reducer.ts [hunk 1]
```diff
@@ -99,6 +100,28 @@ export function reducer(state: State, action: PayloadAction<unknown>): State {
...
+  if (!action.payload) {
+    return state;
+  }
+
+  const { selectedGenes } = state;
+
+  const newSelectedGenes = deleteByItems<State["selectedGenes"][number]>(
+    selectedGenes,
+    // eslint-disable-next-line prettier/prettier
+    [action.payload]
```
[the remark this answers is not in the document]
Comment (ashin-czi) [reply frontend/src/views/WheresMyGene/common/store/reducer.ts]: hmmm it _was_ giving me a linting warning but now it's not for some reason! deleting this line now

**Comment by ashin-czi** [comment]
> I think there's a simpler way to handle the hiding/un-hiding without triggering re-renders. You can add a hover style capture in the parent component that modifies the child. I imagine it would look something like this:
> 
> ```css
> .child-delete-button-class {
>     visibility: hidden;
> }
> .parent-gene-button-class:hover .child-delete-button-class {
>     visibility: visible;
>  }
> ```

Good call! This is fixed now! @seve

**Review by seve — APPROVED** [review]

**Pull request merged**